### ISSUE TITLE
Alternative way to clear search

### ISSUE THREAD

**Issue by jurf** [tpope/vim-sensible#131]
I don't know where I found this but I did quite some research on the topic and it turns out the best way is to do:

    nnoremap <CR> :let @/ = "" <bar> echo "search string cleared whatever" <CR>

This being if you want the mapping to be on `Enter`, which I find really useful

I can't back this up with anything except that it works and doesn't break anything else, but I thought it would be useful.

**Comment by mkropat** [comment]
Setting a `nnoremap <CR>` breaks the [command-line window](http://vimdoc.sourceforge.net/htmldoc/cmdline.html#cmdline-window) and the [quick fix window](http://vimdoc.sourceforge.net/htmldoc/quickfix.html#quickfix-window) (maybe others?), unless [you jump through hoops to prevent it](http://stackoverflow.com/a/11983449/27581).

**Comment by DoctorJellyface** [comment]
It doesn't necessarily have to be `<CR`, it's the `:let @/ = ""` part that's important.

**Comment by justinmk** [comment]
Then n and N won't work.

**Comment by DoctorJellyface** [comment]
...that's the point. Depends on what you prefer really but this seems more intuitive. Also n/N are a bit weird when there isn't any highlighting.

**Comment by ernstki** [comment]
So glad I tucked the URL for this issue in a comment in my `~/.vimrc`, back when I originally remapped `<CR>`!

My QuickFix had been broken by the remapping the whole time, and I didn't even realize it! Thanks to @jurf for the initial tip, and thanks to @mkropat for the save.

Several lessons were learned:
1. if you're going to blindly copy-paste things from the InterWebs, include the link where you found it!
2. don't remap `<CR>` in Vim, or at least be _careful_ about it
    * insights from this [this SO thread](https://stackoverflow.com/a/61793352/785213) were key when I was trying to figure out why [`:Ack`](https://github.com/mileszs/ack.vim) wasn't working as I expected
3. [romainl/vim-cool](https://github.com/romainl/vim-cool) pretty much does what you expect
    * …that is, it clears the `hlsearch` highlights when you switch to another mode, or start using motion commands
    * (tip of the hat to [garybernhardt/dotfiles](https://github.com/garybernhardt/dotfiles/blob/035634543a3a219613fd6b675b57404039b7396c/.vimrc#L12))

Cheers, everyone. I learned something useful the _first_ time I visited this issue back in 20-whatever, and I learned something entirely new and useful the second time I visited it today. It's the issue that keeps on giving. `:)`
